astropy/coordinates/representation.py

```diff
@@ -221,6 +226,138 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def _scale_operation(self, op, *args):
+        results = []
+        for component, cls in self.attr_classes.items():
+            value = getattr(self, component)
+            if issubclass(cls, Angle):
+                results.append(value)
+            else:
+                result = op(value, *args)
+                if result is NotImplemented:
+                    return NotImplemented
+                else:
+                    results.append(result)
+
+        return self.__class__(*results)
+
+    def __mul__(self, other):
+        return self._scale_operation(operator.mul, other)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        return self._scale_operation(operator.truediv, other)
+
+    def __div__(self, other):
+        return self._scale_operation(operator.truediv, other)
+
+    def __neg__(self):
+        return self._scale_operation(operator.neg)
+
+    # Follow numpy convention and make an independent copy.
+    def __pos__(self):
+        return self.__class__(*(getattr(self, component)
+                                for component in self.components), copy=True)
+
+    def __abs__(self):
```

Comment:
The route via `numpy` is in fact substantially slower, mostly because of the creation of the new array. E.g.,
```
sr = SphericalRepresentation(0.*u.deg, 45*u.deg, np.arange(100000.)*u.km)
cr = sr.to_cartesian()
# check Cartesian
%timeit abs(cr)
# 100 loops, best of 3: 3.4 ms per loop
%timeit np.sqrt(np.sum([getattr(cr, component)**2 for component, cls in cr.attr_classes.items() if not issubclass(cls, Angle)], axis=0))
# 100 loops, best of 3: 6.49 ms per loop
# now for Spherical
%timeit abs(sr)
# 1000000 loops, best of 3: 520 ns per loop
%timeit np.sqrt(np.sum([getattr(sr, component)**2 for component, cls in sr.attr_classes.items() if not issubclass(cls, Angle)], axis=0))
# 1000 loops, best of 3: 1.51 ms per loop
```